Fixes #9489 - Explain passthru for `()` expressions (#9526)
MicrosoftDocs/PowerShell-Docs: reference/5.1/Microsoft.PowerShell.Core/About/about_Operators.md

However, in PowerShell, there are additional behaviors.

`(...)` allows you to let output from a _command_ participate in an expression.
For example:

```powershell
PS> (Get-Item *.txt).Count -gt 10
True
```

> [!NOTE]
> Wrapping a command in parentheses causes the automatic variable `$?` to be
> set to `$true`, even when the enclosed command itself set `$?` to `$false`.
> For example, `(Get-Item /Nosuch); $?` unexpectedly yields **True**. For
> more information about `$?`, see
> [about_Automatic_Variables][06].

#### Grouping assignment statements

Ungrouped assignment statements don't output values. When grouping an
assignment statement, the value of the assigned variable is _passed through_
and can be used in larger expressions. For example:

```powershell
PS> ($var = 1 + 2)
3
PS> ($var = 1 + 2) -eq 3
True
```

Wrapping the statement in parentheses turns it into an expression that outputs
the value of `$var`.

This behavior applies to all the assignment operators, including compound

> operators like `+=`, and then increment (`++`) and decrement (`--`) operators.

Reviewer: ```suggestion
operators like `+=`, and the increment (`++`) and decrement (`--`) operators.
```

However, the order of operation for increment and decrement depends on their
position.

```powershell
PS> $i = 0
PS> (++$i) # prefix
1
PS> $i = 0
PS> ($i++) # postfix
0
PS> $i
1
```

In the prefix case, the value of `$i` is incremented before being output. In
the postfix case, the value of `$i` is incremented after being output.

You can also use this technique In the context of a conditional statement, such
as the `if` statement.

```powershell
if ($textFiles = Get-ChildItem *.txt) {
    $textFiles.Count
}
```

In this example, if no files match, the `Get-ChildItem` command returns nothing
and assigns nothing to `$textFiles`, which is considered `$false` in a boolean
context. If one or more **FileInfo** objects are assigned to `$textFiles`, the
conditional evaluates to `$true`. You can work with the value of `$textFiles`

> in the body of the if statement.

Reviewer: ```suggestion
in the body of the `if` statement.
```

    + FullyQualifiedErrorId : CommandNotFoundException
```

The [Invoke-Expression][25] cmdlet can execute code that causes parsing errors
when using the call operator.

```
PS> & "1+1"

> & : The term '1+1' isn't recognized as the name of a cmdlet, function, script

Reviewer: ```suggestion
& : The term '1+1' is not recognized as the name of a cmdlet, function, script
```
